File: backend/app/schemas/gamification.py
"""Pydantic schemas for gamification endpoints."""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class BadgeOut(BaseModel):
    badge_key: str
    earned_at: datetime

    model_config = {"from_attributes": True}


class GamificationSummary(BaseModel):
    total_xp: int
    level: int
    level_name: str
    xp_to_next: Optional[int]  # None when at max level
    streak: int
    badges: list[BadgeOut]

File: src/lib/api/gamification.ts
import { apiFetch } from '@/lib/api';

export interface BadgeOut {
  badge_key: string;
  earned_at: string;
}

export interface GamificationSummary {
  total_xp: number;
  level: number;
  level_name: string;
  xp_to_next: number | null;
  streak: number;
  badges: BadgeOut[];
}

export async function getGamificationSummary(): Promise<GamificationSummary> {
  return apiFetch<GamificationSummary>('/api/v1/gamification/summary');
}
